dali/test/operator_argument.h

```diff
@@ -27,6 +29,48 @@ struct InputArg {
   // TODO(mszolucha): For Dali's argument input
 };
 
+template <typename T, typename Enable = void>
+struct TestOpArgToStringImpl {
+  static std::string to_string(const T&) {
+    return {};
+  }
+};
+
+template <typename T>
+struct TestOpArgToStringImpl<T, typename std::enable_if<std::is_arithmetic<T>::value>::type> {
+  static std::string to_string(const T& val) {
+    return std::to_string(val);
+  }
+};
+
+template <typename T>
+struct TestOpArgToStringImpl<std::vector<T>,
+                             typename std::enable_if<std::is_arithmetic<T>::value>::type> {
+  static std::string to_string(const std::vector<T>& val) {
```

Comment:
You could use `if_iterable` from kernels/util.h and extend it to any collection. Just advertising ;)